File: PolicyPlus/LanguageOptions.vb
Public Class LanguageOptions
    Private OriginalLanguage As String
    Public NewLanguage As String
    Public Function PresentDialog(CurrentLanguage As String) As DialogResult
        OriginalLanguage = CurrentLanguage
        TextAdmlLanguage.Text = CurrentLanguage
        Return ShowDialog()
    End Function
    Private Sub ButtonOK_Click(sender As Object, e As EventArgs) Handles ButtonOK.Click
        Dim selection = TextAdmlLanguage.Text.Trim()
        If selection.Split("-"c).Length <> 2 Then
            MsgBox("Please enter a valid language code.", MsgBoxStyle.Exclamation)
            Exit Sub
        End If
        If selection = OriginalLanguage Then
            DialogResult = DialogResult.Cancel
        Else
            NewLanguage = selection
            DialogResult = DialogResult.OK
        End If
    End Sub
End Class

File: PolicyPlus/LanguageOptions.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace PolicyPlus
{
    public partial class LanguageOptions
    {
        private string OriginalLanguage;
        public string NewLanguage;

        public LanguageOptions()
        {
            InitializeComponent();
        }
        public DialogResult PresentDialog(string CurrentLanguage)
        {
            OriginalLanguage = CurrentLanguage;
            TextAdmlLanguage.Text = CurrentLanguage;
            return ShowDialog();
        }
        private void ButtonOK_Click(object sender, EventArgs e)
        {
            string selection = TextAdmlLanguage.Text.Trim();
            if (selection.Split('-').Length != 2)
            {
                Interaction.MsgBox("Please enter a valid language code.", MsgBoxStyle.Exclamation);
                return;
            }
            if ((selection ?? "") == (OriginalLanguage ?? ""))
            {
                DialogResult = DialogResult.Cancel;
            }
            else
            {
                NewLanguage = selection;
                DialogResult = DialogResult.OK;
            }
        }
    }
}